lib/rules/no-implicit-injections.js

```diff
@@ -0,0 +1,177 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const { getImportIdentifier } = require('../utils/import');
+const Stack = require('../utils/stack');
+const types = require('../utils/types');
+const kebabCase = require('lodash.kebabcase');
+const camelCase = require('lodash.camelcase');
+
+const defaultServiceConfig = { serviceName: 'store', moduleNames: ['Route', 'Controller'] };
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+function fixService(fixer, currentClass, serviceInjectImportName, servicePropertyName) {
+  const serviceInjectPath = kebabCase(servicePropertyName);
+
+  return fixer.insertTextBefore(
+    currentClass.node.body.body[0],
+    `@${serviceInjectImportName}('${serviceInjectPath}') ${servicePropertyName};\n`
+  );
+}
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce usage of implicit service injections',
+      category: 'Deprecations',
+      recommended: false,
+      url: 'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-implicit-injections.md',
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          services: {
+            type: 'array',
+            items: {
+              type: 'object',
+              default: [{ serviceName: 'store', moduleNames: ['Route', 'Controller'] }],
+              properties: {
+                serviceName: {
+                  type: 'string',
+                },
+                moduleNames: {
```

Comment:
Are only specific values allowed for `moduleNames`? If so, we can use `"enum": ["Controller", "Component", ...]` to enforce the allowed values for it.

Reply:
Added enum based on the module name options in the various `isEmber...` in `utils/ember.js`: https://github.com/rtablada/eslint-plugin-ember/blob/rt/no-implicit-injections/lib/utils/ember.js#L275-L317